include/boost/geometry/algorithms/detail/signed_index_type.hpp

```diff
@@ -21,6 +21,7 @@ namespace boost { namespace geometry
 
 
 typedef boost::make_signed<std::size_t>::type signed_index_type;
+typedef boost::make_signed<std::size_t>::type signed_size_type;
```

Comment:
I only changed the names in the buffer right now, since the refactoring of other parts wasn't the purpose of this PR. Therefore I only added a type and didn't rename a file. I thought it'd be better to change the rest in the other PR or to do it directly in the develop since it would be rather automatic. Of course the refactoring of `sectionalize` and posibly other parts of the library would require a PR.